### ISSUE TITLE
getting error when download completed!

### ISSUE THREAD

**Issue by ZCOREP** [bezzad/Downloader#155]
Hello . When the download is done, the download is complete and the file is on the computer, but Downloader_DownloadFileCompleted
It shows an error:
  
Error = {"Unable to read data from the transport connection: An existing connection was forcibly closed by the remote host.."}

Why ? What is the problem ?

**Comment by bezzad** [comment]
Please provide some more information about configs and download URLs so we can do some tests.

**Comment by ZCOREP** [comment]
iam using wpf framework , what config you mean ?

https://dl2.soft98.ir/soft/g/Google.Chrome.119.0.6045.106.x64.zip?1699045797

https://dl2.soft98.ir/soft/m/Mp3tag.3.23.x86.rar?1699046205

**Comment by ZCOREP** [comment]
this is my code
[DownloaderClass.txt](https://github.com/bezzad/Downloader/files/13254956/DownloaderClass.txt)

**Comment by ZCOREP** [comment]
Hello sir, did you checked it?

**Comment by ZCOREP** [comment]
Hello sir, did you checked it?

**Comment by bezzad** [comment]
I tested the given links several times, but I didn't see any problem. Usually, this error occurs when the server or internet network has a problem for a few moments and the TCP connection between your system and the server is interrupted.

**Comment by bezzad** [comment]
I am working on a new version of the downloader, maybe it will give us more information about this problem. This version will be released soon.

**Comment by ZCOREP** [comment]
thanks

**Comment by bezzad** [comment]
@ZCOREP please check the new beta version 3.1.0-beta

**Comment by ZCOREP** [comment]
can i use this version on UWP ?!

**Comment by bezzad** [comment]
@ZCOREP Yes, of course, all platforms can use this package. But in the UWP, you may need to store downloaded files with your direct permissions, use in-memory download methods, get the stream, and then handle it with customized codes.

**Comment by ZCOREP** [comment]
> Hi [@ZCOREP](https://github.com/ZCOREP)
> 
> Please try out the new version of **Downloader v4.0.x beta**. This release includes major updates — we've switched from `HttpWebRequest` to `HttpClient`, which should resolve many previously reported issues.
> 
> Thanks for all your feedback so far — it has been very helpful! If you encounter any new problems, feel free to [open a new issue](https://github.com/bezzad/downloader/issues).
> 
> Looking forward to your feedback!

Hi,
Can I install this new version on UWP as well?
Because the last version you released (v3.3.4) could not be installed on UWP. Thanks!

**Comment by bezzad** [comment]
Hi @ZCOREP,

Yes, the latest version of Downloader (v4.x.x) is compatible with UWP. This update transitions from HttpWebRequest to HttpClient, which is fully supported in UWP applications . This change addresses many of the issues previously encountered with HttpWebRequest in .NET Core.

If you encounter any problems or have further questions, please feel free to let me know!

**Comment by ZCOREP** [comment]
> Hi [@ZCOREP](https://github.com/ZCOREP),
> 
> Yes, the latest version of Downloader (v4.x.x) is compatible with UWP. This update transitions from HttpWebRequest to HttpClient, which is fully supported in UWP applications . This change addresses many of the issues previously encountered with HttpWebRequest in .NET Core.
> 
> If you encounter any problems or have further questions, please feel free to let me know!

Hi,
Thanks for the update!
I downloaded the latest version and tried to add it to my project, but I'm getting an error:
"Unable to add a reference to project 'Downloader'."
I’ve attached a screenshot below for your reference.
Appreciate your help—thanks in advance!


 
![Image](https://github.com/user-attachments/assets/e75d6b87-6f30-4307-aeac-991dd5d4b059)

**Comment by bezzad** [comment]
In v4.x the library is no longer compatible with **.NET Standard / UWP**.

The main reason is that starting from 4.x Downloader moved to a newer HTTP implementation based on `HttpClient` + `SocketsHttpHandler` and uses APIs/features (handler configuration, TLS/SSL options, decompression/range behavior, etc.) that are **not available** (or not supported the same way) in .NET Standard / UWP environments. Because of that, v4.x targets modern .NET runtimes and can’t be consumed reliably by UWP apps.

If you need UWP / .NET Standard compatibility, please use **Downloader v3.x** (for example `v3.3.4`), which still works with those targets.

Unfortunately I can’t maintain/develop the old 3.x line side-by-side with the new 4.x line at the same time, so 4.x will continue focusing on modern .NET.
